plugins/DigitalLibrary/src/DigitalFactoryController.py

```diff
@@ -603,8 +603,8 @@ def saveFileToSelectedProject(self, filename: str, formats: List[str]) -> None:
         self._saveFileToSelectedProjectHelper(filename, formats)
 
     def _saveFileToSelectedProjectHelper(self, filename: str, formats: List[str]) -> None:
-        # Indicate we have started sending a job.
-        self.uploadStarted.emit()
+        # Indicate we have started sending a job (and propagate any user file name changes back to the open project)
+        self.uploadStarted.emit(filename if "3mf" in formats else None)
```

Comment:
This change causes DigitalLibrary to require a higher API version number, meaning it's no longer backwards compatible with Cura 4.9.